Update outerloop workflow

RussKie opened the pull request:
* Tweak the outerloop workflow to align with the interloop runsheet-based workflow. That is, include the "pre-command". Also, add the new GHA test reporter built by @radical. Test run: https://github.com/RussKie/aspire/actions/runs/14984406738
* Explicitly opt-in to parameter randomisation. Resolves #9193
* Restructure playground's AppHostTests
	* Move known flaky/unhealthy tests into own theory. This allows other
tests can be executed in the interloop run, and the flaky to be constrained to the outerloop.
	* Disable SqlServerEndToEnd test, see #9274
	* Add collections to serialise test execution

RussKie committed:
Explicitly opt-in to parameter randomisation

Resolves #9193

RussKie committed:
Update outerloop workflow

radical commented:
Can we trigger a outerloop run for this branch?

radical commented:
Changes look good to me. I just want to confirm with a Outerloop run that my existing tool can still work.

RussKie commented:
Kicked off: https://github.com/dotnet/aspire/actions/runs/14985541555

RussKie committed:
Restructure playground's AppHostTests

* Move known flaky/unhealthy tests into own theory. This allows other
tests can be executed in the interloop run, and the flaky to be constrained
to the outerloop.

* Disable SqlServerEndToEnd test, see #9274

* Add collections to serialise test execution

RussKie commented:
https://github.com/dotnet/aspire/actions/runs/14990633874

radical commented:
Playground tests are still failing. Do we need the playground test changes in this PR?

davidfowl commented:
@radical assuming this is stale and its fine to close this?
